packages/contracts-ops/contracts/CallBatch.sol

```diff
@@ -16,32 +16,93 @@ abstract contract CallBatch is Script {
     using JsonWriter for JsonWriter.Buffer;
     using JsonWriter for string;
 
-    GovernanceMessage.Call[] public calls;
+    GovernanceMessage.Call[] public localCalls;
+    GovernanceMessage.Call[][] public remoteCalls;
+    mapping(string => GovernanceMessage.Call[]) public remoteCallsMap;
 
     bool public complete;
-    string public domain;
+    string public localDomain;
+    string[] public remoteDomains;
     JsonWriter.File outputFile;
 
     function __CallBatch_initialize(
-        string memory _domain,
+        string memory _localDomain,
         string memory _outputFile,
         bool _overwrite
     ) public {
-        require(bytes(domain).length == 0, "already initialized");
+        __CallBatch_initialize(
+            _localDomain,
+            new string[](0),
+            _outputFile,
+            _overwrite
+        );
+    }
+
+    function __CallBatch_initialize(
+        string memory _localDomain,
+        string[] memory _remoteDomains,
+        string memory _outputFilePath,
+        bool _overwrite
+    ) public {
+        require(bytes(localDomain).length == 0, "already initialized");
         require(bytes(outputFile.path).length == 0, "already initialized");
-        domain = _domain;
-        _outputFile = string(abi.encodePacked("./actions/", _outputFile));
-        outputFile.path = _outputFile;
+        outputFile.path = _outputFilePath;
         outputFile.overwrite = _overwrite;
+        localDomain = _localDomain;
+        remoteDomains = _remoteDomains;
```

Comment:
why is this passed at initialize? it should be dynamically updated when remote calls are pushed

Reply:
CallBatch doesn't inherit/use the config, so I assumed this was a design requirement.

Reply:
ah, I see what you mean.